### PR TITLE
Replace use of any in value checker with generic type

### PR DESCRIPTION
**Pull request by mattwynne**
# Description

See details in commit and commit message.

# Motivation & context

Partial fix for #1648

## Type of change

- Refactoring/debt (improvement to code design or tooling without changing behaviour)

# Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING**](../CONTRIBUTING.md) document.
- [ ] I have added tests to cover any changes to behaviour of the code.
- [x] All new and existing tests passed.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ Before anything else, thank you. Thank you for taking some of your precious time
 
 ## Tests
 
-See the `package.json` scripts section for how to run each category of tests.
+Now type `yarn run` or see the `package.json` scripts section for how to run each category of tests.
```
Comment (aurelien-reeves) [inline CONTRIBUTING.md:12]:
```suggestion
```
This is not related to the PR, and another PR suggested the same change

**Review by aurelien-reeves — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (mattwynne) [reply CONTRIBUTING.md:13]: Yeah, I forgot that I had that other change on the same machine and we branched this PR on top of it.

Comment (mattwynne) [reply CONTRIBUTING.md:13]: We should merge that change in and rebase this branch. I don't know why, but the build has been stuck on that other branch.

**Comment by aurelien-reeves** [comment]
> 
> 
> We should merge that change in and rebase this branch. I don't know why, but the build has been stuck on that other branch.

The build shows the following error:

```
 Error : .github#L1
actions/checkout@v2, actions/setup-node@v2, actions/checkout@v2, actions/setup-node@v2, coverallsapp/github-action@master, actions/checkout@v2, and actions/setup-node@v2 are not allowed to be used in cucumber/cucumber-js. Actions in this workflow must be: within a repository owned by cucumber.
```

Someone may have updated the settings of the repo to allow local actions only!?
I suggest to 'allow select actions' including those ones

**Comment by mattwynne** [comment]
Superseded by https://github.com/cucumber/cucumber-js/pull/1764
